integration-tests/ahm/src/tests.rs

```diff
@@ -31,85 +31,21 @@
+use pallet_rc_migrator::types::PalletMigrationChecks;
 use polkadot_runtime::Runtime as Polkadot;
 
 use super::mock::*;
 
 #[tokio::test(flavor = "multi_thread", worker_threads = 2)]
 async fn account_migration_works() {
-	let Some((mut rc, mut ah)) = load_externalities().await else { return };
-	let para_id = ParaId::from(1000);
-
-	// Simulate relay blocks and grab the DMP messages
-	let (dmp_messages, pre_check_payload) = rc.execute_with(|| {
-		let mut dmps = Vec::new();
-
-		if let Ok(stage) = std::env::var("START_STAGE") {
-			let stage = MigrationStage::from_str(&stage).expect("Invalid start stage");
-			RcMigrationStage::<Polkadot>::put(stage);
-		}
-
-		let pre_check_payload =
-			pallet_rc_migrator::preimage::PreimageChunkMigrator::<Polkadot>::pre_check();
-
-		// Loop until no more DMPs are added and we had at least 1
-		loop {
-			next_block_rc();
-
-			let new_dmps =
-				runtime_parachains::dmp::DownwardMessageQueues::<Polkadot>::take(para_id);
-			dmps.extend(new_dmps);
-
-			if RcMigrationStage::<Polkadot>::get() ==
-				pallet_rc_migrator::MigrationStage::MigrationDone
-			{
-				log::info!("Migration done");
-				break (dmps, pre_check_payload);
-			}
-		}
-	});
-	rc.commit_all().unwrap();
-	// TODO: for some reason this prints some small value (2947), but logs on XCM send and receive
-	// show more iteration.
-	log::info!("Num of RC->AH DMP messages: {}", dmp_messages.len());
-
-	// Inject the DMP messages into the Asset Hub
-	ah.execute_with(|| {
-		pallet_ah_migrator::preimage::PreimageMigrationCheck::<AssetHub>::pre_check();
-		let mut fp =
-			asset_hub_polkadot_runtime::MessageQueue::footprint(AggregateMessageOrigin::Parent);
-		enqueue_dmp(dmp_messages);
-
-		// Loop until no more DMPs are queued
-		loop {
-			let new_fp =
-				asset_hub_polkadot_runtime::MessageQueue::footprint(AggregateMessageOrigin::Parent);
-			if fp == new_fp {
-				log::info!("AH DMP messages left: {}", fp.storage.count);
-				break;
-			}
-			fp = new_fp;
-
-			log::debug!("AH DMP messages left: {}", fp.storage.count);
-			next_block_ah();
-
-			if RcMigrationStage::<Polkadot>::get() ==
-				pallet_rc_migrator::MigrationStage::PreimageMigrationDone
-			{
-				pallet_rc_migrator::preimage::PreimageChunkMigrator::<Polkadot>::post_check(
-					pre_check_payload.clone(),
-				);
-			}
-		}
-
-		pallet_ah_migrator::preimage::PreimageMigrationCheck::<AssetHub>::post_check(());
-		// NOTE that the DMP queue is probably not empty because the snapshot that we use contains
-		// some overweight ones.
-	});
+	let Some((rc, ah)) = load_externalities().await else { return };
+	type Payload = <pallet_rc_migrator::preimage::PreimageChunkMigrator<Polkadot> as PalletMigrationChecks>::Payload;
+	let (dmp_messages, pre_check_payload) =
+		rc_migrate::<Payload, pallet_rc_migrator::preimage::PreimageChunkMigrator<Polkadot>>(rc);
+	ah_migrate::<
+		Payload,
+		pallet_rc_migrator::preimage::PreimageChunkMigrator<Polkadot>,
```

Comment:
```suggestion
		_,
```
Not sure if you tried it, but often the Rust compiler knows what you mean